mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp

```diff
@@ -3122,6 +3122,75 @@ transform::VectorizeChildrenAndApplyPatternsOp::applyToOne(
 //===----------------------------------------------------------------------===//
 // VectorizeOp
 //===----------------------------------------------------------------------===//
+ParseResult transform::VectorizeOp::parse(OpAsmParser &parser,
+                                          OperationState &result) {
+  OpAsmParser::UnresolvedOperand target;
+  SmallVector<OpAsmParser::UnresolvedOperand> dynamicSizes;
+  DenseI64ArrayAttr staticSizes;
+  SmallVector<Type> operandTypes;
+  llvm::SMLoc operandLoc;
+  DenseBoolArrayAttr scalableVals;
+
+  if (parser.parseOperand(target) || parser.getCurrentLocation(&operandLoc))
+    return ParseResult::failure();
+
+  if (succeeded(parser.parseOptionalKeyword("vector_sizes"))) {
```

Comment:
Nit: let's have these as `static const StringLiteral kVectorSizesKeyword` somewhere rather than inlined magic constants.

Reply:
i wonder if it's a dumb idea to define these as static class methods the `VectorizeOp`?

Reply:
I'd rather have them as constants local to this file. There is no need for them to be available to all users of `VectorizeOp`, and it would pollute the API and increase linking time.

Reply:
sounds good

Reply:
i made a constant for `vector_sizes` but used `getVectorizeNdExtractAttrName()` for `vectorize_nd_extract`